zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyChecker.java

```diff
@@ -30,31 +31,62 @@ public ForeignKeyChecker(
     enabled = settings.enableForeignKeyChecks();
   }
 
-  private void assertExists(
-      final ZeebeTransaction transaction, final DbForeignKey<? extends DbKey> foreignKey)
+  public void assertExists(
+      final ZeebeTransaction transaction, final ContainsForeignKeys containsForeignKey)
       throws Exception {
+    if (!enabled) {
+      return;
+    }
+    for (final var fk : containsForeignKey.containedForeignKeys()) {
+      assertForeignKeyExists(transaction, fk);
+    }
+  }
+
+  private void assertForeignKeyExists(
+      final ZeebeTransaction transaction, final DbForeignKey<DbKey> foreignKey) throws Exception {
+    if (foreignKey.skip().test(foreignKey.inner())) {
+      return;
+    }
+
     keyBuffer.putLong(0, foreignKey.columnFamily().ordinal(), ZeebeDbConstants.ZB_DB_BYTE_ORDER);
     foreignKey.write(keyBuffer, Long.BYTES);
-    final var value =
-        transaction.get(
-            transactionDb.getDefaultNativeHandle(),
-            transactionDb.getReadOptionsNativeHandle(),
-            keyBuffer.byteArray(),
-            Long.BYTES + foreignKey.getLength());
-    if (value == null) {
+    final var exists =
+        switch (foreignKey.match()) {
+          case Full -> keyExists(transaction, foreignKey);
+          case Prefix -> prefixExists(transaction, foreignKey);
+        };
+    if (!exists) {
```

Comment:
:x: I would suggest to rework the exception thrown. In particular, have different error messages for prefix does not exist and key does not exist. Either by throwing the exception inside the `*Exists()` methods (my preferred way, but then requires renaming of the methods), or by returning `Either` from both methods (which in my mind, is too much memory allocation overhead in a high frequency call site)